### ISSUE TITLE
BigGAN: gradient checkpointing

### ISSUE THREAD

**Issue by gwern** [tensorfork/tensorfork#15]
To scale to 1024px or to use self-attention at higher layers than 64px, memory for a single_n_=1 may exceed the available VRAM on a TPUv3. To pass this limit and train models like GPT-2-1.5b, we can use *gradient checkpointing*: activations inside the model are thrown away to save RAM and then recalculated as necessary for the backwards pass & backprop. This lets a model be much bigger at the cost of some throughput.
